components/brave_wallet/browser/erc_token_registry.h

```diff
@@ -13,27 +13,41 @@
 #include "base/memory/singleton.h"
 #include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
 #include "build/build_config.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+#include "mojo/public/cpp/bindings/receiver_set.h"
+#include "mojo/public/cpp/bindings/remote.h"
 
 namespace brave_wallet {
 
-class ERCTokenRegistry {
+class ERCTokenRegistry : public mojom::ERCTokenRegistry {
  public:
   ERCTokenRegistry(const ERCTokenRegistry&) = delete;
+  ~ERCTokenRegistry() override;
   ERCTokenRegistry& operator=(const ERCTokenRegistry&) = delete;
 
   static ERCTokenRegistry* GetInstance();
+  mojo::PendingRemote<mojom::ERCTokenRegistry> MakeRemote();
   bool ParseERCTokens(const std::string& token_list);
   void UpdateTokenList(std::vector<mojom::ERCTokenPtr> erc_tokens);
   mojom::ERCTokenPtr GetTokenByContract(const std::string& contract);
   mojom::ERCTokenPtr GetTokenBySymbol(const std::string& symbol);
   std::vector<mojom::ERCTokenPtr> GetAllTokens();
 
+  // ERCTokenRegistry interface methods
+  void GetTokenByContract(const std::string& contract,
+                          GetTokenByContractCallback callback) override;
+  void GetTokenBySymbol(const std::string& symbol,
+                        GetTokenBySymbolCallback callback) override;
+  void GetAllTokens(GetAllTokensCallback callback) override;
```

Comment:
Talked to @SergeyZhukovsky and I'll remove the overloaded old calls to `GetTokenByContract`, `GetTokenBySymbol` and `GetAllTokens` soon when I remove the `WalletHandler` code